src/test/java/com/sonyericsson/jenkins/plugins/bfa/BuildFailureScannerHudsonTest.java

```diff
@@ -609,6 +610,120 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher,
+                               BuildListener listener) throws InterruptedException, IOException {
+          build.getWorkspace().child("junit.xml").copyFrom(
+              this.getClass().getResource("junit.xml"));
+          return true;
+        }
+      });
+
+      project.getPublishersList().add(new JUnitResultArchiver("junit.xml", false, null));
+
+      Future<FreeStyleBuild> future = project.scheduleBuild2(0, new Cause.UserIdCause());
+      FreeStyleBuild build = future.get(10, TimeUnit.SECONDS);
+      jenkins.assertBuildStatus(Result.UNSTABLE, build);
+
+      FailureCauseBuildAction action = build.getAction(FailureCauseBuildAction.class);
+      assertNotNull(action);
+
+      List<FoundFailureCause> causeListFromAction = action.getFoundFailureCauses();
+      assertEquals("Amount of failure causes does not match.", 2, causeListFromAction.size());
+
+      // Start with some build description not assigned by BFA and run another scan.
+      action.getFoundFailureCauses();
+      assertEquals(build.getDescription(), null);
+    }
+
+    /**
+     * Test enableBuildDescription = true does append failure cause to existing build description
+     * without categories set.
+     */
+    @Test
+    public void testTestEnableBuildDescriptionWithoutCategoriesIfEnabled() throws Exception {
+        PluginImpl.getInstance().setTestResultParsingEnabled(true);
+        PluginImpl.getInstance().setEnableBuildDescription(true);
+        FreeStyleProject project = jenkins.createFreeStyleProject();
+
+        // Test with a preset build description
+        project.getBuildersList().add(new PrintToLogBuilder(BUILD_LOG));
+        project.getBuildersList().add(new TestBuilder() {
+            public boolean perform(AbstractBuild<?, ?> build, Launcher launcher,
+                                   BuildListener listener) throws InterruptedException, IOException {
+                build.getWorkspace().child("junit.xml").copyFrom(this.getClass().getResource("junit.xml"));
+                build.setDescription("Hello World");
+                return true;
+            }
+        });
+
+        project.getPublishersList().add(new JUnitResultArchiver("junit.xml", false, null));
+
+        Future<FreeStyleBuild> future = project.scheduleBuild2(0, new Cause.UserIdCause());
+        FreeStyleBuild build = future.get(120, TimeUnit.SECONDS);
+        jenkins.assertBuildStatus(Result.UNSTABLE, build);
+
+        FailureCauseBuildAction action = build.getAction(FailureCauseBuildAction.class);
+        assertNotNull(action);
+
+        List<FoundFailureCause> causeListFromAction = action.getFoundFailureCauses();
+        assertEquals("Amount of failure causes does not match.", 2, causeListFromAction.size());
+        String testDescription = "Hello World<br>\n" +
+                "<mark><i>Here are details of the failure...</i>  <i>More details</i></mark>";
+
+        assertEquals(testDescription, build.getDescription());
```

Comment:
Why add test results if you don't have them in the description?

Reply:
They allow a build that fails with some failure causes. It allows the test to make sure the build description is not modified when the feature is disabled.